core/src/com/mygdx/config/GameConfigs.java

```diff
@@ -0,0 +1,56 @@
+package com.mygdx.config;
+
+import com.badlogic.gdx.utils.Array;
+import lombok.NonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class GameConfigs {
+
+    @NonNull
+    private final Map<Class<? extends EntityConfig>, Map<String, EntityConfig>> entityConfigListMap;
```

Comment:
możliwe, że może to być robione przez [ArrayMap](https://libgdx.badlogicgames.com/ci/nightlies/docs/api/com/badlogic/gdx/utils/ArrayMap.html), a nie przez normalną HashMapę/Mapę